### ISSUE TITLE
no-extra-arguments not working with imported functions

### ISSUE THREAD

**Issue by brightsider** [SonarSource/eslint-plugin-sonarjs#431]
**I want to report a bug.**

**Reproducer**

```javascript
// minimal reproducer if relevant
// file A.js
export const createStatus = (params) => UserStatus.create(params);
// file B.js
const userStatusDao = require('./A');
await userStatusDao.createStatus(data, { transaction: t });
```

**Expected behavior**
I need error for this case, but it's ignored.

**eslint-plugin-sonarjs version:** 0.23<br>
**eslint version:** 8.53.0<br>
**Node.js version:** 16.20.2<br>

**Rule key:** no-extra-arguments

**Comment by vdiez** [comment]
Hello @brightsider,

Thanks for reporting this. Indeed, the rule is not currently able to do that. All rules in this repo are "single file" and cannot detect issues like the one you mentioned. 

We'll discuss this improvement internally.

**Comment by frederikprijck** [comment]
Any updates on the plans for this feature? I see it's milestoned as 0.23.0, and 0.23.0 is released without that feature.

**Comment by vdiez** [comment]
Hi @frederikprijck,

sorry for the late response. I moved this ticket to 0.25, since 0.23 and 0.24 did not contain rules hardening.

**Comment by Wohops** [comment]
This issue has been migrated to Jira. [ESLINTJS-28](https://sonarsource.atlassian.net/browse/ESLINTJS-28)
